### PR TITLE
kie-issues#305: Upgrade Kogito Runtimes version to 1.39.0.Final on kie-tools

### LINKED ISSUE #305: Upgrade Kogito Runtimes version to `1.39.0.Final` on `kie-tools`
```[tasklist]
### Acceptance criteria
- [x] Make sure that the Quarkus version is upgraded too to whatever version is compatible with Kogito Runtimes 1.39.0-Final.
- [x] Check for all occurrences of the current version both in `x.x.x.Final` and `x.x.x-Final` formats.
- [ ]
```

### PR DESCRIPTION
**Pull request by thiagoelg**
- Quarkus upgraded to 2.16.7 (`dmn-dev-deployment-quarkus-app`)
- Kogito Runtimes upgraded to 1.39.0-Final (`dmn-dev-deployment-quarkus-app`)
- JITExecutor upgraded to 1.39.0-Final (`extended-services`)
- Added parameters for allowing all origins (CORS), required due to the new changes com Quarkus 2.16.X [1]

[1] https://github.com/quarkusio/quarkus/wiki/Migration-Guide-2.16#no-wildcard-origin-support-by-default-for-cors-filter

Closes https://github.com/kiegroup/kie-issues/issues/305

### CODE REVIEW

**Review by tiagobento — APPROVED** [review]

File: packages/root-env/env/index.js [hunk 1]
```diff
@@ -48,11 +48,11 @@ module.exports = composeEnv([], {
       description: "Enables/disables building example packages during the build.",
     },
     QUARKUS_PLATFORM_version: {
-      default: "2.15.3.Final",
+      default: "2.16.7.Final",
```
Comment (caponetto) [inline packages/root-env/env/index.js:51]: Keep in mind that Quarkus `2.16.7.Final` includes runtimes `1.37.0.Final`.

Comment (tiagobento) [reply packages/root-env/env/index.js:51]: @caponetto So how does that work? Kogito Runtimes depends on Quarkus and Quarkus depends on Kogito Runtimes? :D

Comment (caponetto) [reply packages/root-env/env/index.js:51]: I am not sure how exactly that works but, in practice, you either stick with the runtimes version that comes with the Quarkus you are importing or replace the `quarkus-kogito-bom` with  `kogito-bom` in the `dependencyManagement`.

```xml
<dependency>
    <groupId>org.kie.kogito</groupId>
    <artifactId>kogito-bom</artifactId>
    <version>1.39.0.Final</version>
    <type>pom</type>
    <scope>import</scope>
</dependency>
```

Quarkus `2.16` branch is already with runtimes `1.39.0.Final`, so probably the next release will include this version.

Comment (thiagoelg) [reply packages/root-env/env/index.js:51]: This is really confusing hahaha!
But is there any problem "mixing" theses versions?
The jitexecutor and dmn-dev-deployment-quarkus-app both ran fine 🤔

Comment (caponetto) [reply packages/root-env/env/index.js:51]: Yeah, could different versions lead to unexpected behaviors?!
Anyway, now [kogito-examples](https://github.com/kiegroup/kogito-examples) repository is using Quarkus `2.16.6.Final` and runtimes `1.38.0.Final` through `kogito-bom` (this version of Quarkus includes runtimes `1.36.1.Final`).

**Pull request merged**